text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/placeholder/PlaceholderResolver.java

```diff
@@ -40,88 +38,22 @@
  *
  * @since 4.10.0
  */
+@FunctionalInterface
 public interface PlaceholderResolver {
   /**
-   * Constructs a placeholder resolver from key-value pairs or {@link Placeholder} instances.
+   * Constructs a placeholder resolver from a map.
    *
-   * <p>The {@code pairs} arguments must be a string key followed by a string or {@link ComponentLike} value or a {@link Placeholder}.</p>
+   * <p>
+   *   The provided map is used as the backing for the returned placeholder resolver. This means
+   *   that changes to the map will be reflected in the placeholder resolver.
+   * </p>
    *
-   * @param objects the objects
+   * @param map the map
    * @return the placeholder resolver
    * @since 4.10.0
    */
-  static @NotNull PlaceholderResolver resolving(final @NotNull Object @NotNull ... objects) {
```

Comment:
I liked having this around -- it's a handy shortcut for placeholders.

I see that you moved it into the test classes below as well. If it's useful enough that you want to keep it available for the tests, why is it not useful enough to be in API?